watertap/flowsheets/anaerobic_digester/modified_ADM1_flowsheet.py

```diff
@@ -0,0 +1,344 @@
+        logger=_log,
+        fail_flag=True,
+    )
+
+    return m, results
+
+
+def build(bio_P=False):
+    m = pyo.ConcreteModel()
+
+    m.fs = FlowsheetBlock(dynamic=False)
+
+    # Properties
+    m.fs.props_ASM2D = ModifiedASM2dParameterBlock()
+    m.fs.rxn_props_ASM2D = ModifiedASM2dReactionParameterBlock(
+        property_package=m.fs.props_ASM2D
+    )
+    m.fs.props_ADM1 = ModifiedADM1ParameterBlock()
+    m.fs.props_vap_ADM1 = ADM1_vaporParameterBlock()
+    m.fs.rxn_props_ADM1 = ModifiedADM1ReactionParameterBlock(
+        property_package=m.fs.props_ADM1
+    )
+
+    # Feed water stream
+    m.fs.FeedWater = Feed(property_package=m.fs.props_ASM2D)
+
+    # ======================================================================
+    # Anaerobic digester section
+    # ASM2d-ADM1 translator
+    m.fs.translator_asm2d_adm1 = Translator_ASM2d_ADM1(
+        inlet_property_package=m.fs.props_ASM2D,
+        outlet_property_package=m.fs.props_ADM1,
+        inlet_reaction_package=m.fs.rxn_props_ASM2D,
+        outlet_reaction_package=m.fs.rxn_props_ADM1,
+        has_phase_equilibrium=False,
+        outlet_state_defined=True,
+        bio_P=bio_P,
+    )
+
+    # Anaerobic digester
+    m.fs.AD = AD(
+        liquid_property_package=m.fs.props_ADM1,
+        vapor_property_package=m.fs.props_vap_ADM1,
+        reaction_package=m.fs.rxn_props_ADM1,
+        has_heat_transfer=True,
+        has_pressure_change=False,
+    )
+
+    # ADM1-ASM2d translator
+    m.fs.translator_adm1_asm2d = Translator_ADM1_ASM2D(
+        inlet_property_package=m.fs.props_ADM1,
+        outlet_property_package=m.fs.props_ASM2D,
+        inlet_reaction_package=m.fs.rxn_props_ADM1,
+        outlet_reaction_package=m.fs.rxn_props_ASM2D,
+        has_phase_equilibrium=False,
+        outlet_state_defined=True,
+    )
+
+    # Product Blocks
+    m.fs.Treated = Product(property_package=m.fs.props_ASM2D)
+
+    # ======================================================================
+    # Link units related to AD section
+    m.fs.stream_feed_translator = Arc(
+        source=m.fs.FeedWater.outlet, destination=m.fs.translator_asm2d_adm1.inlet
+    )
+    m.fs.stream_translator_AD = Arc(
+        source=m.fs.translator_asm2d_adm1.outlet, destination=m.fs.AD.inlet
+    )
+    m.fs.stream_AD_translator = Arc(
+        source=m.fs.AD.liquid_outlet, destination=m.fs.translator_adm1_asm2d.inlet
+    )
+    m.fs.stream_translator_product = Arc(
+        source=m.fs.translator_adm1_asm2d.outlet, destination=m.fs.Treated.inlet
+    )
+    pyo.TransformationFactory("network.expand_arcs").apply_to(m)
+
+    return m
+
+
+def set_operating_conditions(m, bio_P=False):
+    # Feed Water Conditions
+    print(f"DOF before feed: {degrees_of_freedom(m)}")
+    m.fs.FeedWater.flow_vol.fix(0.003 * pyo.units.m**3 / pyo.units.s)
+    m.fs.FeedWater.temperature.fix(308.15 * pyo.units.K)
+    m.fs.FeedWater.pressure.fix(1 * pyo.units.atm)
+
+    if bio_P is True:
```

Comment:
From an extremely quick glance, it looks like you are using two different feed compositions depending on whether bio_P is True or False. Other than helping with model convergence, I am not sure why we would do this. In my mind, we would want to be able to observe the difference in results, while using the same feed composition, between applying bio_P=True vs. bio_P=False.

Reply:
The two sources feed composition is corresponding to the the simulation results from `BMS2_P_extension.py`, with `BioP` is `False` or `True`, we get different AD inlet concetrations. I want to test the stablility of the flowsheet with the background of BSM2